Leiningen engine cannot not execute clojure code with dependencies

Currently the leiningen engine can only evaluate clojure code which does not use any external clojure library.
This makes it in practice impossible to use clojure in knitr documents.

```{r engine="lein"}
(use '[clojure.core.matrix :as m])
```
fails with:

```
clojure.lang.Compiler$CompilerException:  Caused by: java.io.FileNotFoundException: Could not locate clojure/core/matrix__init.class or clojure/core/matrix.clj on classpath"
```

The "lein exec" plugin used by the lein engine does allow to use the dependencies of a leiningen project file in the same folder, if present.
This just requires to call "lein exec -ep" instead of "lein exec -e".

To change this would be fully backwards compatible, as a missing "project.clj" file is simply ignored, and so in this case "-ep" it has the same behavior as "-e".

--- a/R/engine.R
+++ b/R/engine.R
@@ -106,7 +106,7 @@ eng_interpreted = function(options) {
       f
     )
   } else paste(switch(
-    engine, bash = '-c', coffee = '-e', groovy = '-e', lein = 'exec -e',
+    engine, bash = '-c', coffee = '-e', groovy = '-e', lein = 'exec -ep',
     mysql = '-e', node = '-e', octave = '--eval', perl = '-e', psql = '-c',
     python = '-c', ruby = '-e', scala = '-e', sh = '-c', zsh = '-c', NULL
   ), shQuote(paste(options$code, collapse = '\n')))

